dev/test/bulk-writer.ts

```diff
@@ -1053,6 +1065,43 @@ describe('BulkWriter', () => {
       });
     return bulkWriter.close().then(() => {
       verifyOpCount(1);
+      expect(timeoutHandlerCounter).to.equal(MAX_RETRY_ATTEMPTS - 1);
+    });
+  });
+
+  it('applies maximum backoff on retries for RESOURCE_EXHAUSTED', async () => {
```

Comment:
Let's also add a test that uses multiple different backoffs in a batch and a test that verifies that we send a second batch before the first batch if the first batch has a backoff.

Reply:
done.